From e0e27f9bc7718925f9b3c9646ddc73f1cc976864 Mon Sep 17 00:00:00 2001
From: Deyan Hadzhiev <deyan.z.hadzhiev@gmail.com>
Date: Fri, 9 May 2014 19:20:24 +0300
Subject: [PATCH] Adds #if checks to the wxRichTextXMLHandler initializations
 for wxUSE_XML to enable builds of richtext without using the expat library.

---
 src/richtext/richtextbuffer.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/richtext/richtextbuffer.cpp b/src/richtext/richtextbuffer.cpp
index ae16861..8b0ac1c 100644
--- a/src/richtext/richtextbuffer.cpp
+++ b/src/richtext/richtextbuffer.cpp
@@ -11405,7 +11405,7 @@ public:
         wxRichTextBuffer::SetRenderer(new wxRichTextStdRenderer);
         wxRichTextBuffer::InitStandardHandlers();
         wxRichTextParagraph::InitDefaultTabs();
-
+#if wxUSE_XML
         wxRichTextXMLHandler::RegisterNodeName(wxT("text"), wxT("wxRichTextPlainText"));
         wxRichTextXMLHandler::RegisterNodeName(wxT("symbol"), wxT("wxRichTextPlainText"));
         wxRichTextXMLHandler::RegisterNodeName(wxT("image"), wxT("wxRichTextImage"));
@@ -11415,7 +11415,7 @@ public:
         wxRichTextXMLHandler::RegisterNodeName(wxT("cell"), wxT("wxRichTextCell"));
         wxRichTextXMLHandler::RegisterNodeName(wxT("table"), wxT("wxRichTextTable"));
         wxRichTextXMLHandler::RegisterNodeName(wxT("field"), wxT("wxRichTextField"));
-
+#endif // wxUSE_XML
         return true;
     }
     void OnExit() wxOVERRIDE
@@ -11423,7 +11423,9 @@ public:
         wxRichTextBuffer::CleanUpHandlers();
         wxRichTextBuffer::CleanUpDrawingHandlers();
         wxRichTextBuffer::CleanUpFieldTypes();
+#if wxUSE_XML
         wxRichTextXMLHandler::ClearNodeToClassMap();
+#endif // wxUSE_XML
         wxRichTextDecimalToRoman(-1);
         wxRichTextParagraph::ClearDefaultTabs();
         wxRichTextCtrl::ClearAvailableFontNames();
-- 
1.9.1

